doc/release/upcoming_changes/27883.change.rst

```diff
@@ -0,0 +1,18 @@
+Changes to the main iterator and potential numerical changes
+------------------------------------------------------------
+The main iterator used in math functions, for ``np.nditer`` and
+on the ``NpyIter`` in the C-API now behaves different for some
+buffered iterations.
+This means that:
+
+* The actual buffer size used will more often be smaller than
+  the maximum buffer sized allowed
+* The "growinner" flag is now honored with buffered reqductions
+  when no operand requires buffering.
+
+At least for ``np.sum()`` such changes in buffersize can change
```

Comment:
Omit "At least".

Maybe: "buffersize may slightly change numerical results due to different chunking of floating point operations"